Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,941 @@
+    ], function(
+        Cartesian2,
+        Cartesian3,
+        ClockRange,
+        ClockStep,
+        Color,
+        createGuid,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        Event,
+        getFilenameFromUri,
+        Iso8601,
+        JulianDate,
+        loadBlob,
+        loadXML,
+        CesiumMath,
+        NearFarScalar,
+        PinBuilder,
+        RuntimeError,
+        TimeInterval,
+        TimeIntervalCollection,
+        HorizontalOrigin,
+        LabelStyle,
+        Autolinker,
+        Uri,
+        when,
+        BillboardGraphics,
+        DataSource,
+        DataSourceClock,
+        Entity,
+        EntityCollection,
+        LabelGraphics,
+        PathGraphics,
+        PolylineGraphics,
+        PolylineOutlineMaterialProperty,
+        SampledPositionProperty) {
+    "use strict";
+
+    var parser = new DOMParser();
+    var autolinker = new Autolinker({
+        stripPrefix : false,
+        twitter : false,
+        email : false,
+        replaceFn : function(linker, match) {
+            if (!match.protocolUrlMatch) {
+                //Prevent matching of non-explicit urls.
+                //i.e. foo.id won't match but http://foo.id will
+                return false;
+            }
+        }
+    });
+
+    var BILLBOARD_SIZE = 32;
+
+    function readBlobAsText(blob) {
+        var deferred = when.defer();
+        var reader = new FileReader();
+        reader.addEventListener('load', function() {
+            deferred.resolve(reader.result);
+        });
+        reader.addEventListener('error', function() {
+            deferred.reject(reader.error);
+        });
+        reader.readAsText(blob);
+        return deferred;
+    }
+
+    function proxyUrl(url, proxy) {
+        if (defined(proxy)) {
+            if (new Uri(url).isAbsolute()) {
+                url = proxy.getURL(url);
+            }
+        }
+        return url;
+    }
+
+    function getOrCreateEntity(node, entityCollection) {
+        var id = queryStringAttribute(node, 'id');
+        id = defined(id) ? id : createGuid();
+        var entity = entityCollection.getOrCreateEntity(id);
+        return entity;
+    }
+
+    function readCoordinate(value) {
+        if (!defined(value)) {
+            return undefined;
+        }
+
+        var digits = value.match(/[^\s,\n]+/g);
+        if (digits.length !== 2 && digits.length !== 3) {
+            window.console.log('GPX - Invalid coordinates: ' + value);
+            return undefined;
+        }
+
+        var longitude = parseFloat(digits[0]);
+        var latitude = parseFloat(digits[1]);
+        var height = parseFloat(digits[2]);
+
+        longitude = isNaN(longitude) ? 0.0 : longitude;
+        latitude = isNaN(latitude) ? 0.0 : latitude;
+        height = isNaN(height) ? 0.0 : height;
+
+        return Cartesian3.fromDegrees(longitude, latitude, height);
+    }
+
+    function readCoordinates(element) {
```

Comment:
`readCoordinates` and `readCoordinate` are no longer used and should be removed.